seacrowd/sea_datasets/nus_sms_corpus/nus_sms_corpus.py

```diff
@@ -0,0 +1,197 @@
+                        },
+                        "userProfile": {
+                            "userID": {"$": datasets.Value("string")},
+                            "age": {"$": datasets.Value("string")},
+                            "gender": {"$": datasets.Value("string")},
+                            "nativeSpeaker": {"$": datasets.Value("string")},
+                            "country": {"$": datasets.Value("string")},
+                            "city": {"$": datasets.Value("string")},
+                            "experience": {"$": datasets.Value("string")},
+                            "frequency": {"$": datasets.Value("string")},
+                            "inputMethod": {"$": datasets.Value("string")},
+                        },
+                    },
+                    "destination": {
+                        "@country": datasets.Value("string"),
+                        "destNumber": {"$": datasets.Value("string")},
+                    },
+                    "messageProfile": {
+                        "@language": datasets.Value("string"),
+                        "@time": datasets.Value("string"),
+                        "@type": datasets.Value("string"),
+                    },
+                    "collectionMethod": {
+                        "@collector": datasets.Value("string"),
+                        "@method": datasets.Value("string"),
+                        "@time": datasets.Value("string"),
+                    },
+                }
+            )
+        elif self.config.schema == _SEACROWD_SCHEMA:
+            features = SCHEMA_TO_FEATURES[TASK_TO_SCHEMA[_SUPPORTED_TASKS[0]]]  # ssp_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        lang = self.config.subset_id
+        data_path = Path(dl_manager.download_and_extract(_URLS[lang]))
+        data_file = list(data_path.glob("*.xml"))[0]
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_file": data_file,
+                },
+            ),
+        ]
+
+    def xml_element_to_dict(self, element: ET.Element) -> Dict:
+        """Converts an xml element to a dictionary."""
+        element_dict = {}
+
+        # add text with key '$', attributes with '@' prefix
+        element_dict["$"] = element.text
+        for attrib, value in element.attrib.items():
+            element_dict[f"@{attrib}"] = value
+
+        # recursively
+        for child in element:
+            child_dict = self.xml_element_to_dict(child)
+            element_dict[child.tag] = child_dict
+
+        return element_dict
```

Comment:
something like this:
```suggestion
    def xml_element_to_dict(self, element: ET.Element, root=True) -> Dict:
        """Converts an xml element to a dictionary."""
        element_dict = {}

        # add text with key '$', attributes with '@' prefix
        if not root: #this is needed since on the feature list we don't have a field key of `$`
            element_dict["$"] = element.text
        for attrib, value in element.attrib.items():
            element_dict[f"@{attrib}"] = value

        # recursively
        for child in element:
            child_dict = self.xml_element_to_dict(child, root=False)
            element_dict[child.tag] = child_dict

        return element_dict
```
```suggestion
    def xml_element_to_dict(self, element: ET.Element) -> Dict:
        """Converts an xml element to a dictionary."""
        element_dict = {}

        # add text with key '$', attributes with '@' prefix
        element_dict["$"] = element.text
        for attrib, value in element.attrib.items():
            element_dict[f"@{attrib}"] = value

        # recursively
        for child in element:
            child_dict = self.xml_element_to_dict(child)
            element_dict[child.tag] = child_dict

        return element_dict
```

Reply:
@sabilmakbar How about, as a simple but ugly workaround, we just add `$` attribute to each key?

Reply:
I actually can't test anything because everything seems to be working on my end, both mac and ubuntu. So I guess I need to pass this to someone.

Reply:
hmm prob the issue wasn't about the platform, but to the `datasets` versions. If I remember it correctly, newer `datasets` version needs assertions of schema generated from `_generate_examples` vs defined in `_info`